site/js/modules/autograder/core.js

```diff
@@ -129,7 +131,11 @@ function sendRequest({
         'body': body,
     });
 
-    return response.then(resolveAPIResponse, resolveAPIError);
+    let usingContextUser = ((override_email == null) && (override_cleartext == null));
```

Comment:
What if someone (on the call endpoint page) enters in a username,
but not a password?
Then, when they auth error wouldn't this bounce them out?

Instead of setting this by examining the arguments,
you may want to just make it an argument to sendRequest().